src/lib/authentication.ts

```diff
@@ -57,4 +58,16 @@ export default class Auth {
     const { data } = jwt.verify(token, signingKey) as any;
     return ChtSession.createFromDataString(data);
   }
+
+  public static async apiAuth(username: string, password: string, domain: string) {
+    const authInfo = Config.getAuthenticationInfo(domain);
+    try {
+      const chtSession = await ChtSession.create(authInfo, username, password);
+      return chtSession;
+    } catch (error: any) {
+      console.error(`Login error: ${error}`);
```

Comment:
this is a bit of a weird function i think. 

1. why does it exist. it just calls ChtSession.create... i think it is largely obfuscating what is happening
2. why are we suppressing exceptions like this? and returning an empty object? weird
3.  logging that there was a "login error"?

can you re-evaluate what's going on and try to clean up?